internal/core/src/bitset/detail/element_wise.h

```diff
@@ -0,0 +1,976 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <optional>
+
+#include "proxy.h"
+
+#include "ctz.h"
+#include "popcount.h"
+
+namespace milvus {
+namespace bitset {
+namespace detail {
+
+// This one is similar to boost::dynamic_bitset
+template <typename ElementT>
+struct ElementWiseBitsetPolicy {
+    using data_type = ElementT;
```

Comment:
using block_type is better

Reply:
I'd like to keep it as is for the consistency with other files.